src/tools/compiletest/src/runtest/parallel.rs

```diff
@@ -0,0 +1,16 @@
+use crate::runtest::{TestCx, WillExecute};
+
+impl TestCx<'_> {
+    pub(super) fn run_parallel_test(&self) {
+        let pm = self.pass_mode();
+        let emit_metadata = self.should_emit_metadata(pm);
+
+        // Repeated testing due to instability in multithreaded environments.
+        let iteration_count = self.props.iteration_count.unwrap_or(50);
+        for _ in 0..iteration_count {
+            let proc_res = self.compile_test(WillExecute::No, emit_metadata);
+            // Ensure there is no ICE during parallel complication.
+            self.check_no_compiler_crash(&proc_res, false);
```

Comment:
I think we should add a timeout here to prevent CI from getting stuck in a deadlock.

Reply:
The current compiletest executor is basically libtest's. That is, tests are ran under test threads. There's a test running for too long *warning message*, but no hard kill-after-timeout, which IIUC is hard to implement with this setup because it'd require some kind of collaborative scheme.